lib/error/lib/merr.c

```diff
@@ -155,7 +155,8 @@ merr_strinfo(
 
         if (ret < 0) {
             /* Try to just return what we already have. */
-            buf[sz] = '\000';
+            if (buf_sz > 0)
+                buf[sz - 1] = '\000';
```

Comment:
@beckerg

Reply:
Funny how long this has existed.

Reply:
LTO compile found it, and some other stuff I'm trying to work through...

Reply:
Do you think it is worth add an LTO test+build for CI after your changes?

Reply:
Yep, that's what I am thinking at the moment.  Once we get it working would be good to keep it that way!

Reply:
Cool. Just ping me after your PR goes in that has the fixes and I will add it